development/extensions/index.md

```diff
@@ -284,9 +360,169 @@ Once installed on the [Onboard Computer](@/hardware/required/onboard-computer/in
+    - The name should be something like `blueos-extension-image-name` (e.g. `blueos-quickstart`)
+3. Set up [GitHub Secrets](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#creating-secrets-for-a-repository):
+    - `DOCKER_USERNAME`: your Docker Hub username
+    - `DOCKER_PASSWORD`: a Docker Hub [access token](https://docs.docker.com/security/for-developers/access-tokens/) (Read & Write), to allow your GitHub repository actions to upload Docker images to your Docker Hub account
+4. Set up [GitHub Variables](https://docs.github.com/en/actions/learn-github-actions/variables#creating-configuration-variables-for-a-repository):
+    - `IMAGE_NAME`: the name you chose for your Docker repository
+    - `MY_EMAIL`: the developer's email address
+    - `MY_NAME`: the developer's name
+    - `ORG_EMAIL`: a support email for the maintaining organisation
+    - `ORG_NAME`: the name of the maintaining organisation
+5. Run the deployment workflow[^4] to automatically build and upload your Docker image (refresh the page to see its status):
+
+{{ easy_image(src="github-workflow-dispatch", width=600, center=true) }}
+
+6. Once the action completes (it may take ~5 minutes) confirm the image is available in your Docker repository
+7. If you have a BlueOS device available, try [manually installing the extension](#manual-development-installs)
+
+[^4]: The QuickStart repositories make use of our [BlueOS Extension Deployment Action](https://github.com/BlueOS-Community/Deploy-BlueOS-Extension), which is available for use in any GitHub workflow, in case you would prefer to use a custom one.
+
+##### Offline Building
+
+While we generally recommend automated builds for convenience and consistency, it is also possible to develop, build, and upload Docker images directly from your development computer.
+
+We recommend using [buildx](https://github.com/docker/buildx?tab=readme-ov-file#buildx) for cross-platform building, and [qemu](https://www.qemu.org) for emulating armv7 on x86-based developer hardware. Our [deployment action](https://github.com/BlueOS-community/Deploy-BlueOS-Extension/blob/main/action.yml) provides our best-practice steps for Extension packaging, so is a useful starting point.
+
+There is also a [basic extension template](https://github.com/BlueOS-community/blueos-extension-template) for Extensions that are intended to be built manually.
+
+{% warning() %}
+Each supported hardware architecture requires its own Docker Image to be built, which our deployment action helps with.
+
+Multi-hardware support allows your Extension to be run/tested on different hardware, but can make development and maintenance more challenging.
+In particular, different hardwares have different build requirements, which may have different library compatibilities.
+
+For reference:
+- Raspberry Pi &le;3 boards use the `arm/v7` architecture
+- Raspberry Pi &ge;4 boards use the `arm64/v8` architecture
```

Comment:
I think we are still using armv7 in the pi4 builds. armv64/v8 should run anything v7 fine, IIRC.
Not sure if that is worth mentioning here.